packages/@guardian/source-foundations/src/typography/fs.ts

```diff
@@ -27,17 +27,18 @@ export const fs: Fs =
 	(category) =>
 	(level, { lineHeight, fontWeight, fontStyle, unit }) => {
 		const fontFamilyValue = fontMapping[category];
-		const fontSizeValue =
+		const fontSizeValue: `${number}rem` | number =
 			unit === 'px'
-				? fontSizeMapping[category][level]
-				: `${remFontSizeMapping[category][level]}rem`;
-		const lineHeightValue =
+				? Number(fontSizeMapping[category][level])
+				: // @ts-expect-error -- the types actually overlap, see https://gist.github.com/mxdvl/5e31fd5b13670b6a41ddac6c65efeee4
```

Comment:
I really cannot get my head around why this is necessary 😅

Reply:
The conversion and division by 16 is not known by the TS compiler. The pixel dimensions as a single integer gets converted to a `number`.

The way around this would be to statically declare pixels to rem for each of the sizes, making this less DRY.

Reply:
I think this gets quite ugly:

<img width="739" alt="Screenshot 2022-01-17 at 11 18 00" src="https://user-images.githubusercontent.com/76776/149760479-80413856-e553-47a9-86b1-f1aeafc57351.png">